packages/terminal/src/common/terminal-protocol.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { JsonRpcServer } from '@theia/core/lib/common/messaging/proxy-factory';
+
+export const ITerminalServer = Symbol('ITerminalServer');
+
+export const terminalPath = '/services/terminal';
+
+export interface ITerminalServer extends JsonRpcServer<ITerminalClient> {
+    create(options: object): Promise<number>;
+    resize(id: number, cols: number, rows: number): Promise<void>;
+}
+
+export interface ITerminalServerOptions {
+    command: string,
+    args?: string[],
+    options?: object
+}
+
+export const ITerminalClient = Symbol('ITerminalClient');
+
+export interface ITerminalExitEvent {
+    terminalId: number;
+    code: number;
+    signal?: number;
+}
+
+export interface ITerminalErrorEvent {
+    terminalId: number;
+    error: Error
```

Comment:
I am not sure that we can pass an instance of `Error` over network, plus `Error` in browser is different from error in node.js

Reply:
If I try to see the defenition of error I get:

````
interface Error {
    name: string;
    message: string;
    stack?: string;
}
```
From lib.es5.d.ts

So isn't Error a typescript built-in ?

Reply:
I mean is `Error` pure JSON object? Won't the info be lost by serialization/deserialization? If you test and it is fine then it is fine with me.

Reply:
It is given the interface I pasted above. Tested it and works properly